models/restore_history_helpers.go

```diff
@@ -44,6 +47,14 @@ func FindRestoreHistoryItems(q *reform.Querier, filters *RestoreHistoryItemFilte
 		args = append(args, filters.ServiceID)
 	}
 
+	if filters != nil && filters.ArtifactID != "" {
+		if _, err := FindArtifactByID(q, filters.ArtifactID); err != nil {
+			return nil, err
+		}
+		conditions = append(conditions, fmt.Sprintf("artifact_id = %s", q.Placeholder(1)))
```

Comment:
does it work when u specify artifactId and also serviceID? I mean because of `q.Placeholder(1)`

Reply:
nice catch, thanks! It worked because I used one filter only 😄